ark/mibi/tiling_utils.py

```diff
@@ -0,0 +1,218 @@
+
+    # need to copy fov metadata over, needed for create_tiled_regions
+    tiling_params['fovs'] = copy.deepcopy(fov_tile_info['fovs'])
+
+    # assign fields to tiling_params
+    tiling_params['region_start_x'] = region_start_x
+    tiling_params['region_start_y'] = region_start_y
+    tiling_params['fov_num_x'] = fov_num_x
+    tiling_params['fov_num_y'] = fov_num_y
+    tiling_params['x_fov_size'] = x_fov_size
+    tiling_params['y_fov_size'] = y_fov_size
+    tiling_params['randomize'] = randomize
+
+    # whether to insert moly points between tiles
+    # NOTE: moly points will be inserted between different runs regardless of what's set here
+    moly_insert = int(
+        input("Specify moly point tile interval? Enter 0 for no and 1 for yes: ")
+    )
+
+    if moly_insert not in [0, 1]:
+        print("Error: moly insertion parameter must enter 0 or 1")
+        moly_insert = int(
+            input("Specify moly point tile interval? Enter 0 for no and 1 for yes: ")
+        )
+
+    # if moly insert is set, we need to specify an additional moly_interval param
+    # NOTE: the interval applies regardless of if the tiles overlap runs or not
+    if moly_insert:
+        moly_interval = int(input("Enter the fov interval size to insert moly points: "))
+
+        while moly_interval < 1:
+            print("Error: moly interval must be positive")
+            moly_interval = int(input("Enter the fov interval size to insert moly points: "))
+
+        tiling_params['moly_interval'] = moly_interval
+
+    return tiling_params
+
+
+def create_tiled_regions(tiling_params, moly_path):
+    """Create the tiled regions for each fov
+
+    Args:
+        tiling_params (dict):
+            The tiling parameters created by set_tiling_params
+        moly_path (str):
+            Path to the JSON moly point file, needed to separate fovs
+
+    Returns:
+        dict:
+            Data containing information about each tile, will be saved to JSON
+    """
+
+    # helper function for creating all pairs between two lists
+    def pairs(*lists):
+        for t in combinations(lists, 2):
+            for pair in product(*t):
+                yield pair
+
+    # file path validation
+    if not os.path.exists(moly_path):
+        raise FileNotFoundError("Moly point file %s does not exist" % moly_path)
+
+    # read in the data
+    with open(moly_path, 'r') as mf:
+        moly_point = json.load(mf)
+
+    # get the current time info
+    dt = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S')
+
+    # define the fov tiling info
+    tiled_regions = {
+        'exportDateTime': dt,
+        'fovFormatVersion': tiling_params['fovFormatVersion'],
+        'fovs': []
+    }
+
+    # define a counter to determine where to insert a moly point
+    # only used if tiling_params['moly_interval'] is set
+    moly_counter = 0
+
+    # append tile information to fov_list, iterate through each tiles iteratively
```

Comment:
This comment could be more clear. We're iterating through each tiled region here, not each tile